flytekit/types/structured/structured_dataset.py

```diff
@@ -334,16 +334,17 @@ class StructuredDatasetTransformerEngine(TypeTransformer[StructuredDataset]):
     Handlers = Union[StructuredDatasetEncoder, StructuredDatasetDecoder]
     Renderers: Dict[Type, Renderable] = {}
 
-    @staticmethod
-    def _finder(handler_map, df_type: Type, protocol: str, format: str):
+    @classmethod
+    def _finder(cls, handler_map, df_type: Type, protocol: str, format: str):
         try:
             return handler_map[df_type][protocol][format]
         except KeyError:
             try:
-                hh = handler_map[df_type][protocol][""]
+                default_format = cls.DEFAULT_FORMATS.get(df_type, PARQUET)
```

Comment:
I don't fully understand this.  why search for parquet?  What happens if we leave it as it is now?

The thinking for all this was - if you request format X, and there's a handler for X then that's what is used.  If format X is not available, then check to see if there's a generic handler.  But I didn't have a good way of specifying what's "generic", so I just picked an empty string.

I think there's a difference between a generic handler, and a default format.  If a user specified a format of Y, but a handler for Y cannot be found, even if the default format is X, I don't think it should be used.  We should look to see if there's a generic (""), and if not then fail.